internal/engine/engine.go

```diff
@@ -49,50 +52,75 @@ func (e *Engine) ExecuteScenario(scenario *Scenario) error {
 }
 
 // Validates a deployment scenario.
-func (e *Engine) TestScenario(scenario *Scenario) error {
+func (e *Engine) TestScenario(scenario *common.Scenario) error {
 	return fs.UsingDirectory(e.Configuration.WorkingDirectory, func() error {
 		az.SetCorrelationId(e.Configuration.CorrelationId, scenario.Environment)
+		stepsToExecute := filterDeletionCommands(scenario.Steps, e.Configuration.DoNotDelete)
+
+		model, err := test.NewTestModeModel(
+			scenario.Name,
+			e.Configuration.Subscription,
+			e.Configuration.Environment,
+			stepsToExecute,
+			lib.CopyMap(scenario.Environment),
+		)
+		if err != nil {
+			return err
+		}
+
+		common.Program = tea.NewProgram(model, tea.WithAltScreen(), tea.WithMouseCellMotion())
+
+		var finalModel tea.Model
+		finalModel, err = common.Program.Run()
+		var ok bool
+
+		// TODO(vmarcella): After testing is complete, we should generate a report.
+
+		model, ok = finalModel.(test.TestModeModel)
```

Comment:
This converts the generic interface `tea.Model` to our specific `test.TestModeModel` struct. `TestModeModel` is a struct which has an implementation for the interface `tea.Model`, but bubble tea (The TUI framework we use), only returns it as a `tea.Model` since that is the only part of the struct that it's concerned with. Casting it back is done so that we can access the properties found on `TestModeModel` after the scenario has completed and bubble tea has closed out the TUI.